src/BitStringAddresses/multicomponent.jl

```diff
@@ -0,0 +1,66 @@
+"""
+    BoseFS2C{NA,NB,M,AA,AB} <: AbstractFockAddress
+
+Address type that constructed with two [`BoseFS{N,M,S}`](@ref). It represents a
+Fock state with two components, e.g. two different species of bosons with particle
+number `NA` from species S and particle number `NB` from species B. The number of
+modes `M` is expected to be the same for both components.
+"""
+struct BoseFS2C{NA,NB,M,SA,SB} <: AbstractFockAddress
+    bsa::BoseFS{NA,M,SA}
+    bsb::BoseFS{NB,M,SB}
+end
+
+BoseFS2C(onr_a::Tuple, onr_b::Tuple) = BoseFS2C(BoseFS(onr_a),BoseFS(onr_b))
+
+function Base.show(io::IO, b::BoseFS2C{NA,NB,M,AA,AB}) where {NA,NB,M,AA,AB}
+    print(io, "BoseFS2C(")
+    Base.show(io,b.bsa)
+    print(io, ",")
+    Base.show(io,b.bsb)
+    print(io, ")")
+end
+
+num_particles(::Type{<:BoseFS2C{NA,NB}}) where {NA,NB} = NA + NB
+num_modes(::Type{<:BoseFS2C{<:Any,<:Any,M}}) where {M} = M
+num_components(::Type{<:BoseFS2C}) = 2
+Base.isless(a::T, b::T) where {T<:BoseFS2C} = isless((a.bsa, a.bsb), (b.bsa, b.bsb))
+
+function near_uniform(::Type{<:BoseFS2C{NA,NB,M}}) where {NA,NB,M}
+    return BoseFS2C(near_uniform(BoseFS{NA,M}), near_uniform(BoseFS{NB,M}))
+end
+
+"""
+    CompositeFS(addresses...) <: AbstractFockAddress
+
+Used to encode addresses for multi-component models.
+"""
```

Comment:
```suggestion
"""
    CompositeFS(addresses::SingleComponentFockAddress...) <: AbstractFockAddress

Used to encode addresses for multi-component models. All component addresses
are expected have the same number of modes `M`.

See also: [`BoseFS`](@ref), [`FermiFS`](@ref), [`SingleComponentFockAddress`](@ref), [`num_modes`](@ref)
"""
```
We should consider exporting `SingleComponentFockAddress`. If we don't we better remove the `(@ref)` links from the docstrings, e.g. for `BoseFS`.